tests/python_package_test/test_engine.py

```diff
@@ -2185,3 +2185,34 @@ def _imptcs_to_numpy(X, impcts_dict):
                     'split_gain', 'threshold', 'decision_type', 'missing_direction',
                     'missing_type', 'weight', 'count'):
             self.assertIsNone(tree_df.loc[0, col])
+
+    def test_interaction_constraints(self):
+        X, y = load_boston(True)
+        num_features = X.shape[1]
+        train_data = lgb.Dataset(X, label=y)
+        # check that constraint containing all features is equivalent to no constraint
+        params = {'verbose': -1,
+                  'seed': 0}
+        est = lgb.train(params, train_data)
+        pred1 = est.predict(X)
+        params = {'interaction_constraints': [list(range(num_features))],
+                  'verbose': -1,
+                  'seed': 0}
+        est = lgb.train(params, train_data)
```

Comment:
I think we can reduce the duplication of code and increase params readability by writing
```
params = {'verbose': -1,
          'seed': 0}
est = lgb.train(params, train_data)
...
est = lgb.train(dict(params,
                     interaction_constraints=[list(range(num_features))]),
                train_data)
...
est = lgb.train(dict(params,
                     interaction_constraints=[list(range(num_features // 2)),
                                              list(range(num_features // 2, num_features))]),
                train_data)
```